tests/appsec/test_blocking_addresses.py

```diff
@@ -638,16 +654,16 @@ class Test_Suspicious_Request_Blocking:
 
     def test_blocking(self):
         """Test if requests that should be blocked are blocked"""
-        assert False, "TODO"
+        pytest.skip("TODO: test not implemented")
```

Comment:
The problem with `assert False` is that it requires that all the languages explicitly skip the test, which seems a bit silly to me, especially because what typically happens next when the test is implemented is that a bunch of libraries will have XPASS test results that take months (I can think of cases where it's more like years) to be tackled. But let me know if I should revert it.

Reply:
> The problem with assert False is that it requires that all the languages explicitly skip the test

It is exactly what we want :) without a proper test, we want to show in the output that the feature is not tested, and thus, we need to explicitly make it fail, and skip it. 

The good decorator for that is `missing_feature`. We could have done a decorator `missing_test`, but we decided a while to not use it, following this  key idea : `test is a part of the feature`.

Reply:
@cbeauchesne saying the test if part of the feature only (arguably) makes sense conceptually. In practice, the feature may be there, exposed to clients and reporting (correctly or incorrectly) data to the backend. The only obvious interpretation of a failure+skip if a missing or incorrect implementation. And by not explicitly skipping the test in every body you would immediately know once the test is implemented where the feature works fine (test passing, rather than XPASS, would indicates an _unexpected_ passing), and where it doesn't (test failing, and then an XFAIL would convey the proper meaning).

That said, this only affects me insofar as I need to explicitly marking all these unimplemented tests as failing. These annotations convey absolutely no useful information.

Reply:
> In practice, the feature may be there, exposed to clients and reporting (correctly or incorrectly) data to the backend.

From testing POV, it does not make a difference. For instance,  "incorrectly" can include "not reported", which is exactly the same behavior as "not implemented". 

> especially because what typically happens next when the test is implemented is that a bunch of libraries will have XPASS test results that take months (I can think of cases where it's more like years) to be tackled

"years" is probably a little bit extreme, especially since we have the easy-wins reports. But anyway, when the test is  implemented, removing the decorators/update the manifest will tells you what are the passing libraries.

Actually, you're right, the current workflow requires : 

* to declare the test as failing using decorator/manifest
* to remove the decorator/manifest when you implement the test 

Which are two extra steps that can be seen as painfull 😄 

But the issue with your solution is that the report won't be correctly interpreted by the feature parity dashbaord (decorator formats the reason with identifiers that must be respected).

So if you really don't want to declare them with the correct way, you can simply removes the test (or totally comment it) : it will achieve what you want (no extra step), and won't mess up the feature parity dashbaord.